src/state_machine/coordinator/fire.rs

```diff
@@ -1352,11 +1325,25 @@ impl<Aggregator: AggregatorTrait> CoordinatorTrait for Coordinator<Aggregator> {
         } else {
             self.current_dkg_id = self.current_dkg_id.wrapping_add(1);
         }
+
         info!("Starting DKG round {}", self.current_dkg_id);
         self.move_to(State::DkgPublicDistribute)?;
         self.start_public_shares()
     }
 
+    /// Check timeout then process a message if passed one
+    fn process(
+        &mut self,
+        packet: &Packet,
+    ) -> Result<(Option<Packet>, Option<OperationResult>), Error> {
+        let (outbound_packet, operation_result) = self.process_timeout()?;
+        if outbound_packet.is_some() || operation_result.is_some() {
+            return Ok((outbound_packet, operation_result));
+        }
+
+        self.process_message(packet)
+    }
```

Comment:
Nice.

Now this raises a question: if `process_timeout` returns `Some(_)` in one of the tuple slots, then should we still process the packet or should we definitely ignore it in all cases? I don't know what's the "proper" thing to do.

Reply:
I *think* the proper thing to do is ignore the packet.  If we allow packets to come in after the timeout, that allows the exact DOS (#70) that this PR is supposed to fix.

Reply:
> Plus if the timeout fires, that means the state has changed so the packet would be dropped anyway if we tried to process it.

Yeah, that's what I was thinking too. The peer who sent this packet could not be in a state where accepting and processing this packet is acceptable, since the coordinator is the one who dictates the pace. Would you mind adding a comment here explaining why dropping the packet is okay.

Reply:
Done